stdlib/public/core/StringHashable.swift

```diff
@@ -14,8 +14,8 @@ import SwiftShims
 
 func _emptyASCIIHashBuffer() -> _UIntBuffer<UInt64, UInt8> {
   var buffer = _UIntBuffer<UInt64, UInt8>()
-  // null terminated strings need to hash differently than non-null terminated
-  // ones. A partially filled ascii buffer should have 1s in the leftover space
+  // we need to take into account the nul suffix for nul-terminated strings.
```

Comment:
Don't we need to also take into account nul suffixes for non-nul-terminated strings? My point is that nothing here has anything to do with treating nul-terminated string differently, and more about not colliding with trailing nuls when hashing.

Reply:
Thanks!
https://github.com/apple/swift/pull/15343